### DISCUSSION TITLE
Clarity in documentation about the managed workflow compatibility

### DISCUSSION THREAD

**Discussion by felixsebastian** [hyochan/expo-iap#19] [Q&A]
Hey, I'm pretty new to react-native so please correct me if I'm wrong.

My understanding is that react-native-iap was **not** compatible with expo managed workflows. From the readme, it seems like this module should work with managed workflows.

If this is the case, it would be good to emphasise this in the docs, as this is a huge upgrade for small teams that don't have the resources to work with bare workflows.

Really appreciate your work here!

**Comment by valentin-debris** [comment]
Hi, 

Currently from my point of view, this repo is not ready to by used yet. I tried it with a small app that only use one time payment (no subscription), it doesn't work on Android, neither on iOS. So, from now, the only solution to use IAP with Expo is RevenueCat. I really want to use this module but for now, it's not possible. I hope @hyochan will provide an upgrade soon, I have another app in the pipe  !

**Comment by see2ever** [comment]
> Hi,
> 
> Currently from my point of view, this repo is not ready to by used yet. I tried it with a small app that only use one time payment (no subscription), it doesn't work on Android, neither on iOS. So, from now, the only solution to use IAP with Expo is RevenueCat. I really want to use this module but for now, it's not possible. I hope [@hyochan](https://github.com/hyochan) will provide an upgrade soon, I have another app in the pipe !

but RevenueCat need it's account, which feels a bit uncomfortable, like being forced into it。

and it would be best to provide a sample code to help the author debug more easily.

**Comment by hyochan** [comment]
Hi @felixsebastian,

Thanks for raising this issue and for your kind words! You’re absolutely right—clarity around compatibility with Expo’s managed workflow is super important, especially for small teams who rely on its simplicity.

### Managed Workflow Compatibility
To clarify: Yes, `expo-iap` is designed to work with Expo’s managed workflow! Unlike `react-native-iap`, which required a bare workflow due to its native dependencies, `expo-iap` is built as an Expo module. This means you can use it without ejecting to a bare setup, starting with Expo SDK 51+ (currently in development). However, to use `expo-iap` in a managed workflow, you’ll need to leverage the **Expo Dev Client**. The Dev Client allows you to include custom native modules like `expo-iap` while still staying within the managed workflow—no full ejection required! This is a huge win for teams avoiding native code complexity, and I agree it’s worth emphasizing in the docs.

I’m currently working on formalizing this in the documentation. You can check out the in-progress version here: [Expo IAP Documentation](https://github.com/hyochan/expo-iap/blob/main/iap.md). It includes installation steps specifically for managed workflows, like adding `"expo-iap"` to the `plugins` array in `app.json` or `app.config.js`, which handles Android permissions and iOS setup automatically. I’ll make sure to add a bold note highlighting this key difference from `react-native-iap`, along with a mention of the Dev Client requirement.

### Current State & Feedback
To address some of the comments from @valentin-debris and @see2ever: I totally get the frustration if things aren’t working as expected yet. This module is still maturing—while it’s functional for basic consumable, non-consumable, and subscription purchases (backed by StoreKit 2 on iOS and Google Play Billing on Android), there are edge cases and bugs (like acknowledgment issues on Android) that need ironing out. For example, some users have reported issues with payments not persisting, which I’m investigating.

For now, it’s usable in simpler apps, but I’d recommend thorough testing before production (RevenueCat remains a solid alternative if you need something fully stable today). That said, I’m actively working on updates to improve reliability—stay tuned!

### Example Code
@see2ever, you’re right that sample code helps with debugging and adoption. I’ve added an example in the repo: [example/App.tsx](https://github.com/hyochan/expo-iap/blob/main/example/App.tsx). It shows how to:
- Initialize the connection (`initConnection`)
- Fetch products and subscriptions (`getProducts`, `getSubscriptions`)
- Handle purchases (`requestPurchase`, `requestSubscription`)
- Listen for updates/errors (`purchaseUpdatedListener`, `purchaseErrorListener`)

This runs in a managed workflow setup with Expo SDK 51 and the Expo Dev Client. Feel free to try it out and let me know if you hit specific issues—I’d love to debug with your input!

### Next Steps
- **Docs Update**: I’ll add a section explicitly stating “Works with Expo Managed Workflow—no bare setup required! (Requires Expo Dev Client)” to the README and `iap.md`.
- **Testing**: I’ll prioritize replicating and fixing the Android/iOS failures mentioned by @valentin-debris. If you can share more details (e.g., error logs, product types), that’d speed things up!
- **Contributions**: PRs to improve docs or code are welcome—especially from small teams like yours who’d benefit from this.

Thanks again for the feedback, and apologies for any confusion so far. Let me know how I can assist further!

**Comment by hyochan** [comment]
Documentation is now live!
https://hyochan.github.io/expo-iap

**Reply by kenny-i** [reply]
@hyochan. "For example, some users have reported issues with payments not persisting, which I’m investigating."

This is still an issue, sometimes connected stays false no matter how much I restart the app, and it becomes true at few random times.

**Comment by alejandromartinez-8** [comment]
I am trying to use it with expo and when i try to do eas build --platform android --profile production --clear-cache  i am getting errors into the build i try a lot of things and i can not do the build can you please assit with this i like to generate a build with subscriptions into the app and i can not do it. I am getting errors with Kotlin versions. I am using expo SDK 53
